apps/auth0-rules/src/__tests__/add-user-metadata.test.ts

```diff
@@ -187,6 +198,47 @@ describe('Auth0 Rule - Add User Metadata', () => {
     });
   });
 
+  it('adds the user metadata on successful fetch for gp2', async () => {
+    nock(apiURL, {
+      reqheaders: {
+        authorization: `Basic ${apiSharedSecret}`,
+      },
+    })
+      .get(`/webhook/users/${user.user_id}`)
+      .reply(200, gp2ApiUser);
+
+    const cb: jest.MockedFunction<Parameters<typeof addUserMetadata>[2]> =
+      jest.fn();
+
+    await addUserMetadata(
+      user,
+      {
+        ...context,
+        request: {
+          query: {},
+          body: { redirect_uri: 'https://gp2.asap.science/' },
+        },
+      },
+      cb,
+    );
+
+    expect(cb).toHaveBeenCalled();
+    const [err, resUser, resContext] = cb.mock.calls[0];
+    expect(err).toBeFalsy();
+    expect(resUser).not.toBeNull();
+    expect(resContext).not.toBeNull();
+    expect(resContext.idToken['https://gp2.asap.science/user']).toStrictEqual({
```

Comment:
it might be a good idea to extract the expectation to a variable and type it - it's a good practice because that way any time the type changes we know which expectations need to be updated. You would think jest would just make sure that whatever is in .toEqual(...) should adhere to the type inside expect() but i think it does not have to

Reply:
Done.